commit d6b09e754c23b657544f6e7f39fbf7de24c58aa2
Author: Akinobu Mita <akinobu.mita@gmail.com>
Date:   Tue Jul 31 16:41:31 2012 -0700

    fault-injection: fix failcmd.sh warning
    
    "fault-injection: add tool to run command with failslab or
    fail_page_alloc" added tools/testing/fault-injection/failcmd.sh to make it
    easier to inject slab/page allocation failures by fault injection.
    
    failcmd.sh prints the following warning when running with arguments
    for command.
    
            # ./failcmd.sh echo aaa
            failcmd.sh: line 209: [: echo: binary operator expected
            aaa
    
    This warning is caused by an improper check whether at least one
    parameter is left after parsing command options.
    
    Fix it by testing the length of $1 instead of $@
    
    Signed-off-by: Akinobu Mita <akinobu.mita@gmail.com>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

diff --git a/tools/testing/fault-injection/failcmd.sh b/tools/testing/fault-injection/failcmd.sh
index 1776e924b202..78a9ed7fecdb 100644
--- a/tools/testing/fault-injection/failcmd.sh
+++ b/tools/testing/fault-injection/failcmd.sh
@@ -206,7 +206,7 @@ while true; do
 	esac
 done
 
-[ -z "$@" ] && exit 0
+[ -z "$1" ] && exit 0
 
 echo $oom_kill_allocating_task > /proc/sys/vm/oom_kill_allocating_task
 echo $task_filter > $FAULTATTR/task-filter